tests/components/bluesound/test_config_flow.py

```diff
@@ -0,0 +1,254 @@
+"""Test the Bluesound config flow."""
+
+from unittest.mock import AsyncMock
+
+import voluptuous as vol
+
+from homeassistant.components.bluesound.const import DOMAIN
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER, SOURCE_ZEROCONF
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+async def test_user_flow_success(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_player_sync_status: AsyncMock
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+    assert result["step_id"] == "user"
+    assert result["data_schema"] == vol.Schema(
+        {
+            vol.Required(CONF_HOST): str,
+            vol.Optional(CONF_PORT, default=11000): int,
+        }
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: "1.1.1.1",
+        },
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "player-name"
+    assert result["data"] == {CONF_HOST: "1.1.1.1", CONF_PORT: 11000}
+
+    mock_setup_entry.assert_called_once()
+    mock_player_sync_status.assert_called_once()
+
+
+async def test_user_flow_cannot_connect(
+    hass: HomeAssistant, mock_player_sync_status_client_connection_error: AsyncMock
+) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: "1.1.1.1",
+        },
+    )
+
+    assert result["type"] is FlowResultType.FORM
```

Comment:
We should test if the flow is able to recover an error, so rewatch and make sure we finish the flow